app/javascript/src/components/editor/CodeMirror.svelte

```diff
@@ -216,12 +217,13 @@
     return true
   }
 
-  function getIntendForLine(state, line) {
-    line = Math.max(line, 0)
+  function getIndentForLine(state, line, charLimit) {
+    let lineText = state.doc.lineAt(Math.max(line, 0)).text
+    lineText = charLimit !== undefined ? lineText.slice(0, charLimit) : lineText
 
-    const spaces = /^\s*/.exec(state.doc.lineAt(line).text)?.[0].length
-    const tabs = /^\t*/.exec(state.doc.lineAt(line).text)?.[0].length
-    return Math.floor((spaces - tabs) / 4) + tabs
+    const tabs = /^\t*/.exec(lineText)?.[0].length
+    const spaces = /^\s*/.exec(lineText)?.[0].length - tabs
+    return Math.floor(spaces / 4) + tabs
```

Comment:
I do feel like this was a little more readable before with the spaces subtracted in the return
```suggestion
    const tabs = /^\t*/.exec(lineText)?.[0].length
    const spaces = /^\s*/.exec(lineText)?.[0].length
    return Math.floor((spaces - tabs) / 4) + tabs
```

Reply:
oh i changed it for the sake of making a bit more readable lol. i dont like how the 'spaces' constant is set to a wrong value just so you can make it correct by subtracting in the return. but the result is the same so you can use the previous if you want!

Reply:
Oh true, tabs are included in the regex. Fair point!